src/shogun/machine/LinearMachine.cpp

```diff
@@ -120,3 +141,50 @@ void CLinearMachine::store_model_features()
 {
 }
 
+float64_t CLinearMachine::compute_bias(CFeatures* data)
+{
+	REQUIRE(m_labels,"No labels set\n");
+
+	if (!data)
+		data=features;
+
+	REQUIRE(data,"No features set\n");
+
+	REQUIRE(m_labels->get_num_labels() == data->get_num_vectors(),"Number of training vectors does not match number of labels\n");
+
+	SGVector<float64_t> outputs = apply_get_outputs(data);
+
+	int32_t num_vec=data->get_num_vectors();
+
+	Map<VectorXd> eigen_outputs(outputs,num_vec);
+	Map<VectorXd> eigen_labels(((CRegressionLabels*)m_labels)->get_labels(),num_vec);
+
+	return (eigen_labels - eigen_outputs).mean() ;
+}
+
+
+bool CLinearMachine::train(CFeatures* data)
+{
+	/* not allowed to train on locked data */
+	if (m_data_locked)
+	{
+		SG_ERROR("%s::train data_lock() was called, only train_locked() is"
+				" possible. Call data_unlock if you want to call train()\n",
+				get_name());
+	}
+
+	if (train_require_labels())
+	{
+		REQUIRE(m_labels,"No labels given",this->get_name());
+
+		m_labels->ensure_valid(get_name());
+	}
+
+	bool result = train_machine(data);
+
+	/** optional compute bias*/
+	if(m_compute_bias)
```

Comment:
This doesnt make sense. You computed the mean above, and now you might not store it???